pkg/api/handlers/libpod/containers_create.go

```diff
@@ -33,6 +35,41 @@ func CreateContainer(w http.ResponseWriter, r *http.Request) {
 		utils.InternalServerError(w, err)
 		return
 	}
+
+	for i, d := range sg.Devices {
```

Comment:
As opposed to trying to parse `d.Path` - maybe we can add an extra field to the struct? Make a `specgen.Device` struct that embeds the actual Device struct, and has a `DevicePath` field that contains the path of the device that should be mounted into the container? That means we don't have to parse permissions and destination path, as they can be populated by the user sending the request.

Reply:
@mheon I went with parsing as it was "less" invasive.

Compatibility (I was going to work on this later) uses:
```
Array [
    PathOnHost | string
    PathInContainer | string
    CgroupPermissions | string
]
```
If adding field(s) to the struct, I would add these and use them if set otherwise use existing fields.